packages/atlas-service/src/main.ts

```diff
@@ -559,6 +570,29 @@ export class AtlasService {
     }
   }
 
+  static async signOut(): Promise<void> {
+    // Reset and recreate event emitter first so that we don't accidentally
+    // react on any old plugin instance events
+    this.oidcPluginLogger.removeAllListeners();
+    this.oidcPluginLogger = new EventEmitter();
+    this.attachOidcPluginLoggerEvents();
+    // Destroy old plugin and setup new one
+    await this.plugin?.destroy();
+    this.setupPlugin();
+    // Revoke tokens. Revoking refresh token will also revoke associated access
+    // tokens
+    // https://developer.okta.com/docs/guides/revoke-tokens/main/#revoke-an-access-token-or-a-refresh-token
+    await this.revoke({ tokenType: 'refresh_token' });
```

Comment:
Do you want to try/catch this one so that the rest of the function continues regardless of whether token revocation failed?

Reply:
Yes, we probably should, we remove this token from everywhere anyway and probably not much we can do if we somehow got a network error on revoke

Reply:
Added handling for this in https://github.com/mongodb-js/compass/pull/4742/commits/8bc331b4eecfffd2cff855df3806db8fa43f1cb7